app/assets/javascripts/Components/Result/summary_panel.jsx

```diff
@@ -11,13 +12,66 @@ export class SummaryPanel extends React.Component {
     graceTokenDeductions: []
   };
 
+  markDataSet = {
+    label: I18n.t('activerecord.models.mark.one'),
+    backgroundColor: 'rgba(58,106,179,0.35)',
+    borderColor: '#3a6ab3',
+    borderWidth: 1,
+    hoverBackgroundColor: 'rgba(58,106,179,0.75)'
+  };
+
+  // Colors for chart are based on constants.css file, with modifications for opacity.
+  oldMarkDataSet = {
+    label: I18n.t('results.remark.old_mark'),
+    backgroundColor: 'rgba(250,253,170,0.65)',
+    borderColor: '#dde426',
+    borderWidth: 1,
+    hoverBackgroundColor: '#dde426'
+  };
+
   constructor(props) {
     super(props);
     this.state = {
       showNewExtraMark: false,
+      xTitle: I18n.t('activerecord.models.criterion.one'),
+      yTitle: I18n.t('activerecord.models.mark.one') + ' (%)',
+      datasets: [],
+      labels: [],
+      chartLegend: false
     }
   }
 
+  componentDidMount() {
+    this.marks_modal = new ModalMarkus('#marks_chart');
+  }
+
+  toggleMarksChart = () => {
+    let labels = [];
+    let markData = [];
+    let oldMarks = [];
+    let oldMarksExist = Object.keys(this.props.old_marks).length > 0;
+    this.props.marks.forEach(m => {
+      labels.push(m.name);
+      markData.push(Math.round(m.mark * 100) / m.max_mark);
+      if (oldMarksExist) {
+        if (this.props.old_marks[m.criterion_type + '-' + m.id]) {
```

Comment:
This check is a bit strange. I think that if this is a remark result, then all of the relevant `old_marks` should be in this object, and we should rely on this. So we shouldn't need to have this check here, nor the else branch (which I don't think makes sense as the right behaviour anyway).